wp-includes/js/service-worker.js

```diff
@@ -144,14 +201,12 @@ wp.serviceWorker = workbox;
 	wp.serviceWorker.routing = Object.assign( new WPRouter(), publicAPI );
 }
 
-// @todo There is another 'fetch' handler being for DefaultRouter added in the workbox-routing module which will be unused since.
-self.addEventListener( 'fetch', event => {
-	const request = event.request;
-	const responsePromise = wp.serviceWorker.routing.handleRequest( { request, event } );
-	if ( responsePromise ) {
-		event.respondWith( responsePromise );
-	}
-} );
+/*
+ * Note: The fetch event listener was moved from here to wp-includes/js/service-worker-add-routing-fetch-event-listener.js
+ * The reason for this is that the fetch event needs to be added _after_ the call to precaching.addRoute() is made, so
+ * that precached assets will be handled before falling back to other caching strategies for non-precached routes.
+ * If the caching strategy handler is added first, then the precache route would never handle it.
+ */
```

Comment:
@jeffposnick I wanted to raise this issue that we encountered. It seems that there is a conflict between the precaching and routing modules. It seems the precaching route should always come first, right? At the moment, however, the `fetch` for routing gets added immediately without the ability for something else to come before, right?

https://github.com/GoogleChrome/workbox/blob/b146d8d3c2db89fa51afc322049b9795894bb4f2/packages/workbox-routing/_default.mjs#L199-L206

Reply:
If you call (the wrapped equivalent of) `workbox.precache.precacheAndRoute()` earlier in the service worker file than you call `workbox.routing.registerRoute()`, the precaching should take precedence.

So yes, ordering does matter—we [just updated the docs](https://github.com/google/WebFundamentals/pull/6820/files#diff-a298f19f420180849fe4a3cde57504bfR72) to clarify that.

Are you blocked at something particular that you can't resolve via reordering?

Reply:
Thanks. I've fixed the ordering in 7572d0d15b75fa4e76bfe3038272ec8504a9312f. It was due in part to us using a custom router which logged a notice when multiple matches were made. I suppose I didn't have to eliminate the custom router as I did there to fix the ordering, but it seemed better to do for now since the custom router meant duplicating too much code.